hibernate-core/src/main/java/org/hibernate/jpa/internal/enhance/EnhancingClassTransformerImpl.java

```diff
@@ -23,9 +26,12 @@
 public class EnhancingClassTransformerImpl implements ClassTransformer {
 
 	private final EnhancementContext enhancementContext;
+	private final BytecodeProvider bytecodeProvider;
 
 	public EnhancingClassTransformerImpl(EnhancementContext enhancementContext) {
```

Comment:
There is something weird going on.

This constructor is only ever called from here:

https://github.com/hibernate/hibernate-orm/blob/a986a3806a2756b73845a2e9942b5ea86b0893fd/hibernate-core/src/main/java/org/hibernate/jpa/boot/internal/PersistenceUnitInfoDescriptor.java#L111-L114

But then all implementations of `PersistenceUnitInfo#addTransformer` are no-ops...

This component is probably not being tested. I'd suggest tring your patch in WildFly before merging?

Reply:
Good idea, yes WildFly has a similar implementation but it's a different one.